lib/src/command/add.dart

```diff
@@ -319,20 +328,32 @@ Specify multiple sdk packages with descriptors.''');
       sdkConstraints: original.sdkConstraints,
       dependencies: dependencies,
       devDependencies: devDependencies,
-      dependencyOverrides: original.dependencyOverrides.values,
+      dependencyOverrides: dependencyOverrides,
     );
   }
 
   /// Split [arg] on ':' and interpret it with the flags in [argResult] either as
   /// an old-style or a new-style descriptor to produce a PackageRef].
   _ParseResult _parsePackage(String arg, ArgResults argResults) {
     var isDev = argResults['dev'] as bool;
-    if (arg.startsWith('dev:')) {
-      if (argResults.isDev) {
-        usageException("Cannot combine 'dev:' with --dev");
+    var isOverride = false;
+    final firstColon = arg.indexOf(':');
```

Comment:
Let's do some better parsing here... So we don't have `dev:^1.2.3` acting as something that is added as dev-dependency... it's just weird.

If we try to match: `(?:(dev|override):)?(${identifierRegExp.pattern})(?::(.*))?$` then we'll have:
 * group 1: dev or override
 * group 2: package name
 * group 3: constraint / JSON

If it doesn't match this regexp, we can try to match: `(?:([a-z][a-z0-9_-]):)?(${identifierRegExp.pattern})(?::(.*))?$`, and if it matches, then we can suggest that the modifier in group 1, should be either `dev` or `override`.

Or we can just stick to say what the form is, instead of trying to match a second regex with the only purpose being that we provide a nicer error message.

Reply:
Done